### DISCUSSION TITLE
Run command in the background

### DISCUSSION THREAD

**Discussion by TechStudent10** [tauri-apps/tauri#7737] [Q&A] [answered]
Hey Tauri community! I have a Tauri command that runs a shell command. That shell command, however, takes some time to run. Is there anyway I can run this in the background and return the result to the frontend when it finishes?

**Comment by FabianLars** [comment, marked as answer]
If you mark the command as async either via `async fn` or `#[tauri::command(async)]` it won't block the main thread but you can still return data from the function as usual.
Alternatively you can [spawn](https://doc.rust-lang.org/std/thread/fn.spawn.html) a new thread in the command and use [tauri events](https://tauri.app/v1/guides/features/events/) to send data to the frontend.

**Reply by TechStudent10** [reply]
Oh I didn't know about the async function approach. Thank you! Currently I've sufficed with sending events to and from the front-end, but yeah this will help out a ton
